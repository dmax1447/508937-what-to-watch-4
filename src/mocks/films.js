
const films = [
  {
    id: `f01`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: `f02`,
    title: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
  },
  {
    id: `f03`,
    title: `Macbeth`,
    picture: `macbeth.jpg`,
  },
  {
    id: `f04`,
    title: `Aviator`,
    picture: `aviator.jpg`,
  },
  {
    id: `f05`,
    title: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: `f06`,
    title: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
  },
  {
    id: `f07`,
    title: `Revenant`,
    picture: `revenant.jpg`,
  },
  {
    id: `f08`,
    title: `Johnny English`,
    picture: `johnny-english.jpg`,
  },
];

export default films;
