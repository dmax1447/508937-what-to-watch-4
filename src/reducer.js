import {extend} from "./utils.js";

const initialState = {
  genre: `All_genres`,
};

const ActionType = {
  SET_GENRE: `SET_GENGE`,
  SET_ALL_GENERES: `SET_ALL_GENERES`,
};

const ActionCreator = {
  setGenre: (genre) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  setAllGeneres: () => ({
    type: ActionType.SET_ALL_GENERES,
    payload: null,
  }),
};

const genres = [
  `Drama`,
  `Adventure`,
  `Bio`,
  `Mystery`,
  `Horror`,
  `Comedy`,
];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case ActionType.SET_ALL_GENERES:
      return extend(state, {
        genre: `All_genres`,
      });
  }

  return state;
};


export {reducer, ActionType, ActionCreator, genres};
