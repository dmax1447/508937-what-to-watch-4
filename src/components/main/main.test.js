import React from "react";
import renderer from "react-test-renderer";
import Main from "./main.jsx";

import promo from '../../mocks/tests/promo.js';
import films from '../../mocks/tests/films.js';
import {createStore} from "redux";
import {Provider} from "react-redux";
import {reducer} from "../../reducer.js";

const store = createStore(reducer);

const onCardTitleClick = (evt) => {
  evt.preventDefault();
};

const genres = [...new Set(films.map((item) => item.genre))];

it(`<Main /> should render correctly`, () => {
  const main = renderer
    .create(
        <Provider store={store}>
          <Main
            promo={promo}
            films={films}
            onCardTitleClick={onCardTitleClick}
            genres={genres}
          />
        </Provider>
    ).toJSON();

  expect(main).toMatchSnapshot();
});
