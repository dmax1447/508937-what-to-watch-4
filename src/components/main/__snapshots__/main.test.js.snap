// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
<div>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Birds
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              horror
            </span>
            <span
              className="movie-card__year"
            >
              1963
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Adventure
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Bio
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Mystery
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f01"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="bohemian-rhapsody.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f02"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Macbeth"
              height="175"
              src="macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f03"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Aviator"
              height="175"
              src="aviator.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f04"
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="We need to talk about Kevin"
              height="175"
              src="we-need-to-talk-about-kevin.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f05"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="what-we-do-in-the-shadows.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f06"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Revenant"
              height="175"
              src="revenant.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f07"
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Johnny English"
              height="175"
              src="johnny-english.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/f08"
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
